string request
---
int32 return_code